### DISCUSSION TITLE
Translational symmetry in RBM

### DISCUSSION THREAD

**Discussion by abukva** [netket/netket#1266] [Q&A] [answered]
Hi, I've encountered a bit of a simple problem, but it could also be that I misunderstood something. I am trying to simulate a simple 1D Transverse Ising at the critical point and I would like to run larger system sizes, like a couple of hundreds. My idea was that if I use RBMSymm with translational symmetry that would greatly simplify the problem and I could run larger systems. But when I do introduce symmetries there isn't any noticeable difference in the run time or convergence speed. Am I missing something?

Simple code I am running:

 ```
N = 20
g = nk.graph.Hypercube(length=N, n_dim=1, pbc=True)
hi = nk.hilbert.Spin(s=1/2, N=g.n_nodes)
H = nk.operator.Ising(hilbert=hi, h = 1, J=-1, graph = g)
ma = nk.models.RBMSymm(g.translation_group(), alpha=1.)
sa = nk.sampler.MetropolisLocal(hilbert=hi, n_chains=32)
op = nk.optimizer.Sgd(learning_rate=0.05)
sr = nk.optimizer.SR(diag_shift=0.1)
vs = nk.vqs.MCState(sa, ma, n_samples=4800)
gs = nk.VMC(hamiltonian=H, optimizer=op, preconditioner=sr,  variational_state=vs)
gs.run(out=None, n_iter=400)
```

**Comment by PhilipVinc** [comment, marked as answer]
Hi @abukva , 

> My idea was that if I use RBMSymm with translational symmetry that would greatly simplify the problem

Depends what you mean by `Simplifying the problem`. You still are trying to solve a quantum mechanic hamiltonian with exponentially many degrees of freedom `2^N`, and you are removing a number of `N` degrees of freedom, so that's really little.

Mainly, you are constraining the NQS to be symmetric, and you know the ground state is symmetric, which can speed up convergence and/or yield better results.

> and I could run larger systems

Not really. Computational and memory cost will be comparable. Mainly you get better results. 

> there isn't any noticeable difference in the run time or convergence speed

Runtime, you should not expect speedups. 
Convergence speed, it's problem-dependent. It really depends on the optimiser and learning rate. 
On such a simple problem like 1D Ising.

In short: Symmetries lead to better (more accurate) results, but they don't, in general, speedup calculations or convergence.
